src/index.ts

```diff
@@ -227,51 +231,61 @@ export default class BrowserTransport implements LinkTransport {
+                const infoTitle = this.createEl({class: 'title', tag: 'span', text: 'Success!'})
+                const subtitle = request.isIdentity() ? 'Identity signed.' : 'Transaction signed.'
+                const infoSubtitle = this.createEl({class: 'subtitle', tag: 'span', text: subtitle})
+                infoEl.appendChild(infoTitle)
+                infoEl.appendChild(infoSubtitle)
+                emptyElement(this.requestEl)
+                this.requestEl.appendChild(logoEl)
+                this.requestEl.appendChild(infoEl)
+                this.show()
+                this.closeTimer = setTimeout(() => {
+                    this.hide()
+                }, 1.5 * 1000)
+            } else {
+                emptyElement(this.requestEl)
                 this.hide()
-            }, 1.5 * 1000)
+            }
         }
     }
 
     public onFailure(request: SigningRequest, error: Error) {
         if (request === this.activeRequest) {
             this.clearTimers()
-            this.setupElements()
-            const infoEl = this.createEl({class: 'info'})
-            const logoEl = this.createEl({class: 'logo'})
-            logoEl.classList.add('error')
-            const infoTitle = this.createEl({
-                class: 'title',
-                tag: 'span',
-                text: 'Transaction error',
-            })
-            const infoSubtitle = this.createEl({
-                class: 'subtitle',
-                tag: 'span',
-                text: error.message || String(error),
-            })
-            infoEl.appendChild(infoTitle)
-            infoEl.appendChild(infoSubtitle)
-            emptyElement(this.requestEl)
-            this.requestEl.appendChild(logoEl)
-            this.requestEl.appendChild(infoEl)
-            this.show()
-            this.closeTimer = setTimeout(() => {
+            if (this.requestStatus) {
+                this.setupElements()
+                const infoEl = this.createEl({class: 'info'})
+                const logoEl = this.createEl({class: 'logo'})
+                logoEl.classList.add('error')
+                const infoTitle = this.createEl({
+                    class: 'title',
+                    tag: 'span',
+                    text: 'Transaction error',
+                })
+                const infoSubtitle = this.createEl({
+                    class: 'subtitle',
+                    tag: 'span',
+                    text: error.message || String(error),
+                })
+                infoEl.appendChild(infoTitle)
+                infoEl.appendChild(infoSubtitle)
+                emptyElement(this.requestEl)
+                this.requestEl.appendChild(logoEl)
+                this.requestEl.appendChild(infoEl)
+                this.show()
+                this.closeTimer = setTimeout(() => {
+                    this.hide()
+                }, 5 * 1000)
+            } else {
+                emptyElement(this.requestEl)
```

Comment:
same here

Reply:
Got it - and changed this and the other one